src/Microsoft.DotNet.CodeAnalysis/Analyzers/PinvokeAnalyzer.cs

```diff
@@ -0,0 +1,157 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.IO;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Text;
+using System.Diagnostics;
+
+namespace Microsoft.DotNet.CodeAnalysis.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class PinvokeAnalyzer : DiagnosticAnalyzer
+    {
+        private const string Title = "Invalid Pinvoke call";
+        private const string MessageFormat = @"{0} is not supported on one\more targeted platforms.{1}";
+        private const string Description = "The Pinvoke call should be removed.";
+        private const string AnalyzerName = "PinvokeAnalyzer";
+
+        private static DiagnosticDescriptor InvalidPinvokeCall = new DiagnosticDescriptor(DiagnosticIds.BCL0100.ToString(), Title, MessageFormat, AnalyzerName, DiagnosticSeverity.Error, isEnabledByDefault: true, description: Description);
```

Comment:
nit: BCL0020 should be fine. Trying not to bump the number of warnings by too much